Sources/MapboxCoreNavigation/PassiveLocationManager.swift

```diff
@@ -105,21 +105,37 @@ open class PassiveLocationManager: NSObject {
     
     var lastRawLocation: CLLocation?
     
+    /**
+     A closure to be called when manual location was successfully set.
+     
+     - parameter success: Boolean value, which is set to `true` in case if manual location was
+     successfully set, otherwise `false`.
+     */
+    public typealias UpdateLocationCompletionHandler = (_ success: Bool) -> Void
+    
     /**
      Manually sets the current location.
      
      This method stops any automatic location updates.
+     
+     - parameter location: Location, which will be used by navigator.
+     - parameter completion: Completion handler, which will be called when asynchronous operation completes.
      */
-    public func updateLocation(_ location: CLLocation?) {
+    public func updateLocation(_ location: CLLocation?, completion: UpdateLocationCompletionHandler? = nil) {
         guard let location = location else { return }
         systemLocationManager.stopUpdatingLocation()
         systemLocationManager.stopUpdatingHeading()
-        self.didUpdate(locations: [location])
+        
+        didUpdate(locations: [location]) { success in
+            completion?(success)
+        }
     }
 
-    private func didUpdate(locations: [CLLocation]) {
+    private func didUpdate(locations: [CLLocation], completion: UpdateLocationCompletionHandler? = nil) {
         for location in locations {
-            navigator.updateLocation(for: FixLocation(location))
+            navigator.updateLocation(for: FixLocation(location)) { success in
+                completion?(success)
```

Comment:
Should we call completion for every location in the array?

Reply:
Good point, probably should just call it once. Should it be considered successful only if all the updates succeed, or only if the one of them succeeds?

Reply:
To be serious I'm not sure what would be better here. Looks like completion only in case when all locations were processed would be preferable, as user might not really care about every location in particular. BTW this method is only called from two places: first one is `CLLocationDelegate` and we do not use that completion at all there, second one is in `PassiveLocationManager.updateLocation(_:completion:)` and we provide only one location there, instead of collection.

Reply:
OK, if we ever do need a `PassiveLocationManager.updateLocations(_:completion:)`, then we can revisit this logic at that time.